lib/tpm_hash.c

```diff
@@ -97,3 +97,45 @@ int tpm_hash_compute_data(TSS2_SYS_CONTEXT *sapi_context, BYTE *buffer,
     free(bufferList);
     return rval == TPM_RC_SUCCESS ? 0 : -3;
 }
+
+//
+// This function does a hash on an array of data strings and re-uses syscontext
+//
+UINT32 tpm_hash_sequence(TSS2_SYS_CONTEXT *sapi_context, TPMI_ALG_HASH hash_alg,
+        size_t num_buffers, TPM2B_DIGEST *buffer_list, TPM2B_DIGEST *result) {
+
+    TPMT_TK_HASHCHECK validation;
+    TPMS_AUTH_COMMAND cmd_auth = { 0 };
+    TPMS_AUTH_COMMAND *cmd_session_array[1] = { &cmd_auth };
+    TSS2_SYS_CMD_AUTHS cmd_auth_array = { 1, &cmd_session_array[0] };
+    TPMI_DH_OBJECT sequence_handle;
+    TPM2B_AUTH null_auth = { .t.size = 0 };
+    TPM2B empty_buffer = { .size = 0 };
+
+    // Set result size to 0, in case any errors occur
+    result->b.size = 0;
+
+    // Init input sessions struct
+    cmd_auth.sessionHandle = TPM_RS_PW;
```

Comment:
Can we move this to line 108, something like:

cmd_auth = {
 .sessionHandle = TPM_RS_PW,
 .nonce = { 0 },
 .sessionAttributes = { 0 },
 .hmac = { 0 }
}

Tweak as needed to get it to build, perhaps consider creating a TPM_AUTH_COMMAND_INIT() macro, because this code is pretty boiler plate and needs to be duplicated in a lot of places that initialize this structure.